app/models/schema/common.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+##
+# Defining a common superset of attributes for Resources
+module Schema
+  module Common
+    extend ActiveSupport::Concern
+
+    ##
+    # Retrieve the attributes from Hyrax, imported from external systems, and local
+    def self.attributes
+      Plum.attributes + Schema::IIIF::Book.attributes
+    end
+
+    included do
+      Common.attributes.each do |field|
+        attribute field
```

Comment:
Agreed, something similar was what I was considering for #80 , but I was wary of further blocking this issue.